### ISSUE TITLE
report state change of other responders in a scene

### ISSUE THREAD

**Issue by jbarneson** [TD22057/insteon-mqtt#150]
Happy to be corrected if I misunderstood this supported feature: 

> Report state changes for any device in an Insteon scene when the scene is triggered.

I have an insteon network of 120+ devices, including 15 keypads. I have three "scene" scenarios in my setup. 

1) I have a "night mode" scene that almost every light in the house responds to and most keypads have a button to control/respond. The modem is also aware through a custom scene on the modem.

2) Light groups. Many of my switches and lights are tied together in simulated three ways using scenes. I configured this through Houselinc when i first setup the network. 

3) Fans, vents, and bathroom heaters. I have numerous fanlincs and relays that control bathroom vents and heaters that are tied back to a keypad scene as responders.

In all three scenarios, I am able to control individual devices from insteon-mqtt just fine.. both as a responder and a controller. But other devices in the group/scene do not report updated status back. I know this is a known limitation of the serial network, but my interpretation of insteon-mqtt's documentation was that the service would know of these connections through local copies of the database and would "Report state changes for any device in an Insteon scene when the scene is triggered"

I attempted to relink the responders/controllers from insteon-mqtt and refresh the databases, but that resulted in a timeout. Perhaps a different issue, but i thought the logs might yield insight:

2019-05-19 16:29:04 INFO Mqtt: MQTT message insteon3/command/43.96.c2 b'{ "cmd" : "linking" }'
2019-05-19 16:29:04 UI Mqtt: Commanding switch device 43.96.c2 (hvac_ff_masterbath_heater) cmd=linking
2019-05-19 16:29:04 INFO Base: Device 43.96.c2 (hvac_ff_masterbath_heater) link mode grp 1
2019-05-19 16:29:04 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 09 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:29:04 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 09 01 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 f6 
2019-05-19 16:29:04 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 09 01
2019-05-19 16:29:04 UI Mqtt: Entered linking mode
2019-05-19 16:29:17 INFO Mqtt: MQTT message insteon3/command/modem b'{ "cmd" : "linking" }'
2019-05-19 16:29:17 UI Mqtt: Commanding Modem device 31.18.5b (modem) cmd=linking
2019-05-19 16:29:17 INFO Protocol: Write message to modem: Modem linking: grp: 1 Cmd.EITHER
2019-05-19 16:29:17 INFO Protocol: Read 0x64: Modem linking: grp: 1 Cmd.EITHER ack: True
2019-05-19 16:29:17 UI Mqtt: Modem entering linking mode
2019-05-19 16:29:19 INFO Protocol: Read 0x53: All link done: 43.96.c2 grp: 1 Cmd.RESPONDER cat: 0x00 0x00 0x00
2019-05-19 16:29:19 UI ModemLinkComplete: Modem db entry added via linking: 43.96.c2 grp 1 RESP
2019-05-19 16:29:19 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT cmd: 01 01
2019-05-19 16:29:19 WARNING Protocol: No read handler found for message type 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT cmd: 01 01
2019-05-19 16:29:47 INFO Mqtt: MQTT message insteon3/command/43.96.c2 b'{ "cmd" : "refresh" }'
2019-05-19 16:29:47 UI Mqtt: Commanding switch device 43.96.c2 (hvac_ff_masterbath_heater) cmd=refresh
2019-05-19 16:29:47 INFO Base: Device 43.96.c2 (hvac_ff_masterbath_heater) cmd: status refresh
2019-05-19 16:29:47 INFO Protocol: Write message to modem: Std: 43.96.c2, Type.DIRECT, 19 00 
2019-05-19 16:29:47 INFO Protocol: Read 0x62: Std: 43.96.c2, Type.DIRECT, 19 00 ack: True
2019-05-19 16:29:47 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 05 00
2019-05-19 16:29:47 UI Switch: Switch 43.96.c2 (hvac_ff_masterbath_heater) refresh on=False
2019-05-19 16:29:47 INFO Switch: Setting device 43.96.c2 (hvac_ff_masterbath_heater) on False normal
2019-05-19 16:29:47 INFO Switch: MQTT received on/off 43.96.c2 (hvac_ff_masterbath_heater) on: False normal
2019-05-19 16:29:47 UI DeviceRefresh: Device 43.96.c2 db out of date (got 5 vs None), refreshing
2019-05-19 16:29:47 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:29:47 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:29:48 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 2f 00
2019-05-19 16:29:48 INFO DeviceDbGet: 43.96.c2 device ACK response
2019-05-19 16:29:54 WARNING Base: Handler timed out 1 of 3 sent: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:29:54 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:29:54 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:30:00 WARNING Base: Handler timed out 2 of 3 sent: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:00 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:00 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: False 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:30:00 ERROR DeviceDbGet: 43.96.c2 NAK response
2019-05-19 16:30:06 WARNING Base: Handler timed out 3 of 3 sent: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:06 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:06 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:30:09 INFO Protocol: Read 0x51: Ext: 43.96.c2->31.18.5b Type.DIRECT ext cmd: 2f 00
00 01 0f ff 00 aa 06 32 ff a6 ff 1c 01 1f 
2019-05-19 16:30:09 UI DeviceDbGet: Entry: 0fff: 32.ff.a6 grp:   6 type: RESP data: 0xff 0x1c 0x01
2019-05-19 16:30:15 WARNING Base: Handler timed out - no more retries (3 sent)
2019-05-19 16:30:15 ERROR Mqtt: Command timed out
2019-05-19 16:30:15 INFO Protocol: Read 0x51: Ext: 43.96.c2->31.18.5b Type.DIRECT ext cmd: 2f 00
00 01 0f e7 00 aa 00 31 18 5b 00 1c 01 6f 
2019-05-19 16:30:15 WARNING Protocol: No read handler found for message type 0x51: Ext: 43.96.c2->31.18.5b Type.DIRECT ext cmd: 2f 00
00 01 0f e7 00 aa 00 31 18 5b 00 1c 01 6f 
2019-05-19 16:30:41 INFO Mqtt: MQTT message insteon3/command/43.96.c2 b'{ "cmd" : "refresh", "force" : true }'
2019-05-19 16:30:41 UI Mqtt: Commanding switch device 43.96.c2 (hvac_ff_masterbath_heater) cmd=refresh
2019-05-19 16:30:41 INFO Base: Device 43.96.c2 (hvac_ff_masterbath_heater) cmd: status refresh
2019-05-19 16:30:41 INFO Protocol: Write message to modem: Std: 43.96.c2, Type.DIRECT, 19 00 
2019-05-19 16:30:41 INFO Protocol: Read 0x62: Std: 43.96.c2, Type.DIRECT, 19 00 ack: True
2019-05-19 16:30:42 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 05 00
2019-05-19 16:30:42 UI Switch: Switch 43.96.c2 (hvac_ff_masterbath_heater) refresh on=False
2019-05-19 16:30:42 INFO Switch: Setting device 43.96.c2 (hvac_ff_masterbath_heater) on False normal
2019-05-19 16:30:42 INFO Switch: MQTT received on/off 43.96.c2 (hvac_ff_masterbath_heater) on: False normal
2019-05-19 16:30:42 UI DeviceRefresh: Device 43.96.c2 db out of date (got 5 vs None), refreshing
2019-05-19 16:30:42 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:42 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 05 00
2019-05-19 16:30:42 INFO Protocol: Ignored duplicate Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 05 00
2019-05-19 16:30:42 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:30:43 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 2f 00
2019-05-19 16:30:43 INFO DeviceDbGet: 43.96.c2 device ACK response
2019-05-19 16:30:49 WARNING Base: Handler timed out 1 of 3 sent: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:49 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:49 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:30:55 WARNING Base: Handler timed out 2 of 3 sent: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:55 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:30:55 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: False 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:30:55 ERROR DeviceDbGet: 43.96.c2 NAK response
2019-05-19 16:31:01 WARNING Base: Handler timed out 3 of 3 sent: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:31:01 INFO Protocol: Write message to modem: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
2019-05-19 16:31:01 INFO Protocol: Read 0x62: Ext: 43.96.c2, Type.DIRECT ext, 2f 00 ack: True 00 00 00 00 00 00 00 00 00 00 00 00 00 d1 
2019-05-19 16:31:04 INFO Protocol: Read 0x50: Std: 43.96.c2->31.18.5b Type.DIRECT_ACK cmd: 2f 00
2019-05-19 16:31:04 INFO DeviceDbGet: 43.96.c2 device ACK response
2019-05-19 16:31:04 INFO Protocol: Read 0x51: Ext: 43.96.c2->31.18.5b Type.DIRECT ext cmd: 2f 00
00 01 0f ff 00 aa 06 32 ff a6 ff 1c 01 1f 
2019-05-19 16:31:04 UI DeviceDbGet: Entry: 0fff: 32.ff.a6 grp:   6 type: RESP data: 0xff 0x1c 0x01
2019-05-19 16:31:10 WARNING Base: Handler timed out - no more retries (3 sent)
2019-05-19 16:31:10 ERROR Mqtt: Command timed out

**Comment by TD22057** [comment]
As the docs indicate, you must have a successful link and refresh before the library will work.  That's how it knows to report what's going on - it needs a current copy of the link database.  You can see the device db files in the config directory (they're in a human readable form) so you can check them manually.  You didn't say what kind of devices you can't refresh - that's important.  Battery devices aren't awake and don't respond to messages so you need to wake them up with a sensor trip or activate button and then quickly refresh to get the db downloaded.  This may take several tries to get it to work (but only needs to be done once usually).

Focus on one device at a time.  What kind of device is it?  See if the db file exists.  You can also run print-db from the command line to dump the db that insteon-mqtt has (or just look at the file in the config/ dir).  If it's not a battery device and you can't get refresh to work, send me as much info as you can (device cat, sub cat) by running the get-model command line call.

**Comment by jbarneson** [comment]
Appreciate the response -- and happy to hear this should be functioning as I interpreted the docs. Your documentation is thorough, thank you.

None of my insteon devices are battery operated, so that's not a concern for me. Agreed, I started to narrow down to just one test device that's a simple use case:
I have a 6 btn keypad with one of the scene buttons controlling and responding to a in-line relay (for a bathroom heater). 

When I look in my config folder (config/insteon-mqtt), I don't see any database files in there.. i don't think I ever have. All i have is the config.yaml file and some backups i've made over time.

![image](https://user-images.githubusercontent.com/4008317/58389649-f4a7e300-7ff1-11e9-88a8-50df7044112d.png)

**Comment by TD22057** [comment]
Look inside the config file. It says where to write the db files.

**Comment by jbarneson** [comment]
That pointed me in the right direction. Thank you! 

I updated my config file which was putting the db files in 'data'. I figured out this and the logging directory were relative to the hassio home, so I assume this should have been here: config/data, but there was no data directory on my box. I changed the value to:

  storage: '/config/insteon-mqtt/dbs'

Did a quick test on the device we mentioned above and it worked perfectly. I'm going to roll through the rest of my devices over the coming days. Appreciate your pointer!
